crates/aptos-telemetry/src/build_information.rs

```diff
@@ -1,32 +1,98 @@
 // Copyright (c) Aptos
 // SPDX-License-Identifier: Apache-2.0
 
+use crate as aptos_telemetry;
 use crate::{service::TelemetryEvent, utils};
-use shadow_rs::shadow;
 use std::collections::BTreeMap;
 
 /// Build information event name
 const APTOS_NODE_BUILD_INFORMATION: &str = "APTOS_NODE_BUILD_INFORMATION";
 
 /// Build information keys
-const BUILD_BRANCH: &str = "build_branch";
-const BUILD_CARGO_VERSION: &str = "build_cargo_version";
-const BUILD_CHAIN_ID: &str = "build_chain_id";
-const BUILD_CLAP_VERSION: &str = "build_clap_version";
-const BUILD_COMMIT_HASH: &str = "build_commit_hash";
-const BUILD_OS: &str = "build_os";
-const BUILD_PKG_VERSION: &str = "build_pkg_version";
-const BUILD_PROJECT_NAME: &str = "build_project_name";
-const BUILD_RUST_CHANNEL: &str = "build_rust_channel";
-const BUILD_RUST_VERSION: &str = "build_rust_version";
-const BUILD_TAG: &str = "build_tag";
-const BUILD_TARGET: &str = "build_target";
-const BUILD_TARGET_ARCH: &str = "build_target_arch";
-const BUILD_TIME: &str = "build_time";
-const BUILD_VERSION: &str = "build_version";
+pub const BUILD_BRANCH: &str = "build_branch";
+pub const BUILD_CARGO_VERSION: &str = "build_cargo_version";
+pub const BUILD_CHAIN_ID: &str = "build_chain_id";
+pub const BUILD_CLAP_VERSION: &str = "build_clap_version";
+pub const BUILD_COMMIT_HASH: &str = "build_commit_hash";
+pub const BUILD_OS: &str = "build_os";
+pub const BUILD_PKG_VERSION: &str = "build_pkg_version";
+pub const BUILD_PROJECT_NAME: &str = "build_project_name";
+pub const BUILD_RUST_CHANNEL: &str = "build_rust_channel";
+pub const BUILD_RUST_VERSION: &str = "build_rust_version";
+pub const BUILD_TAG: &str = "build_tag";
+pub const BUILD_TARGET: &str = "build_target";
+pub const BUILD_TARGET_ARCH: &str = "build_target_arch";
+pub const BUILD_TIME: &str = "build_time";
+pub const BUILD_VERSION: &str = "build_version";
 
-// Get access to BUILD information
-shadow!(build);
+// Used by external crates to collect crate specific build information
+#[macro_export]
+macro_rules! collect_build_information {
+    () => {{
+        // Get access to shadow BUILD information
+        shadow_rs::shadow!(build);
+
+        // Collect and return the build information
+        let mut build_information: std::collections::BTreeMap<String, String> = BTreeMap::new();
+        build_information.insert(
+            aptos_telemetry::build_information::BUILD_BRANCH.into(),
```

Comment:
Oh, I just figured one macro is cleaner than two. It's still being used in this crate (as well as in the CLI crate).

Reply:
You can cut down on a bunch of duplicated stuff if you put in the block here `use::aptos_telemetry::build_information::*`, which will clean it up since you're reimporting crate as `aptos_telemetry`

Reply:
Thanks Greg! I think almost all of this is going to be rewritten before mainnet, so just gonna unblock it for now and let whoever is responsible for it in the future handle the code 😄